src/LinkDotNet.Blog.Infrastructure/Persistence/Sql/DbContextInitializer.cs

```diff
@@ -0,0 +1,47 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using System.Linq;
+
+namespace LinkDotNet.Blog.Infrastructure.Persistence.Sql;
+public sealed partial class DbContextInitializer
```

Comment:
Probably we want to have something like a `ManualResetEventSlim` here (via DI container) that gets injected.
In a `finally` block of `Initialize` we would `Set` the `ManualResetEvent` - basically allowing everyone to pass that was waiting via the `Wait` method.

The other two services (BackgroundServices) might wanna use said service to wait for the dbinitializer to finish.

Reply:
On second thought - I am more and more leaning toward your first approach: Calling the migration manually - eliminating much of the code and complexity.

Folks could integrate the migration inside their CI/CD pipeline. 
What are your feelings towards this @Kamyab7?

Of course, if we don't do it automatically, we have to extend the documentation on how to achieve a migration, but that shouldn't be a big issue.

Reply:
> Probably we want to have something like a `ManualResetEventSlim` here (via DI container) that gets injected. In a `finally` block of `Initialize` we would `Set` the `ManualResetEvent` - basically allowing everyone to pass that was waiting via the `Wait` method.
> 
> The other two services (BackgroundServices) might wanna use said service to wait for the dbinitializer to finish.

another thought, I am not sure but I think since we call InitializeStorageProvider before app.Run() background services don't start yet.

```
        //HERE
        app.InitializeStorageProvider(app.Configuration);

        app.Run();
```

Reply:
> On second thought - I am more and more leaning toward your first approach: Calling the migration manually - eliminating much of the code and complexity.
> 
> Folks could integrate the migration inside their CI/CD pipeline. What are your feelings towards this @Kamyab7?
> 
> Of course, if we don't do it automatically, we have to extend the documentation on how to achieve a migration, but that shouldn't be a big issue.


In my opinion, using CI/CD pipeline would be great.

Reply:
Well then we can remove the whole automatic migration part, keeping the code very very lean.